lib/puma/request.rb

```diff
@@ -266,14 +271,12 @@ def normalize_env(env, client)
         uri = URI.parse(env[REQUEST_URI])
         env[REQUEST_PATH] = uri.path
 
-        raise "No REQUEST PATH" unless env[REQUEST_PATH]
-
         # A nil env value will cause a LintError (and fatal errors elsewhere),
         # so only set the env value if there actually is a value.
         env[QUERY_STRING] = uri.query if uri.query
       end
 
-      env[PATH_INFO] = env[REQUEST_PATH]
+      env[PATH_INFO] = env[REQUEST_PATH].to_s # #to_s in case it's nil
```

Comment:
See above comment. PATH_INFO should always be a string.

Reply:
PATH_INFO - sure. but in the tested scenatio REQUEST_PATH is not a string but nil. To make sure we always assign a string (even if it's empty) to PATH_INFO, I'm calling `to_s`